config/routes.rb

```diff
@@ -141,6 +141,10 @@
 
     get "/tags" => "tags#index"
     get "/tags.json" => "tags#index", :format => "json"
+    get "/tags/new" => "tags#new", :as => "new_tag"
```

Comment:
Convert this to `resource` instead of listing actions individually.

Reply:
As I feared when I did it that way, the resourceful route generator really wants to name the route I named `update_tag` as `tag`. However, the existing routes already use the `tag` route name for the `home#tagged` route, which is used all over the site. So to use that, I'd have to change the name of that route, and every reference to that route throughout the app. This feels like something that ought to be done as part of a general refactor to use more resourceful routes throughout the app.

Reply:
I'd expect the resource route to be `POST` or `PUT` instead of `GET`. Rails puts [multiple actions at the same URL](http://guides.rubyonrails.org/routing.html#crud-verbs-and-actions) as long as they have different verbs.

Reply:
Sorry if I didn't describe it well enough. What's happening is the following routes:

```
tag GET      /t/:tag(.:format)                             home#tagged {:format=>/html|rss|json/}
    GET      /t/:tag/page/:page(.:format)                  home#tagged {:format=>/html/}
```

Get created by the following declarations in `routes.rb`:

```
get "/t/:tag" => "home#tagged", :as => "tag", :format => /html|rss|json/
get "/t/:tag/page/:page" => "home#tagged"
```

While adding `resources :tags, except: [:show, :destroy]` generates the routes:

```
       tags GET      /tags(.:format)                               tags#index {:format=>/html/}
            POST     /tags(.:format)                               tags#create {:format=>/html/}
    new_tag GET      /tags/new(.:format)                           tags#new {:format=>/html/}
   edit_tag GET      /tags/:id/edit(.:format)                      tags#edit {:format=>/html/}
            PATCH    /tags/:id(.:format)                           tags#update {:format=>/html/}
            PUT      /tags/:id(.:format)                           tags#update {:format=>/html/}
            GET      /tags.json(.:format)                          tags#index {:format=>/json/}
```

If the above declarations are commented out, then you get the resources routes like follows:

```
       tags GET      /tags(.:format)                               tags#index {:format=>/html/}
            POST     /tags(.:format)                               tags#create {:format=>/html/}
    new_tag GET      /tags/new(.:format)                           tags#new {:format=>/html/}
   edit_tag GET      /tags/:id/edit(.:format)                      tags#edit {:format=>/html/}
        tag PATCH    /tags/:id(.:format)                           tags#update {:format=>/html/}
            PUT      /tags/:id(.:format)                           tags#update {:format=>/html/}
            GET      /tags.json(.:format)                          tags#index {:format=>/json/}
```

The `tag` route identifier is getting assigned to the heavily-used `/t/:tag` route which connects to `home#tagged`, and so is not available for the `/tags/:id` route, which connects to `tags#update`. I don't see any way to change the route name generated by `resources`, so the only way to resolve the conflict and allow the use of `resources` routes is to change the name of the other tag routes, which are used in several other places in the application.

Reply:
Oof, thanks for breaking that down. Too much magic and convention, let's definitely just stick with the way you've got it now.